### PR TITLE
New VPN Protocol from 2018-06-01 REST specs

### PR DESCRIPTION
**Pull request by henry416**
## Description

Added new VPN protocol from 2018-06-01 REST spec (checked in May 24th)
https://github.com/Azure/azure-rest-api-specs/pull/3133

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by henry416** [comment]
@markcowl  When will SDK be generated for June 15 release?

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 1]
```diff
@@ -563,7 +563,7 @@
     </None>
     <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.VirtualNetworkGatewayTests\TestVirtualNetworkGatewayP2SAndSKU.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: remove all changes to this file

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 2]
```diff
@@ -60,6 +60,15 @@
     <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.0.2-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: remove all these changes

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs [hunk 3]
```diff
@@ -149,7 +149,8 @@ public class NewAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
             HelpMessage = "The list of P2S VPN client tunneling protocols")]
         [ValidateSet(
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs]: Change this from a ValidateSet to an ArgumentCompleter

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs [hunk 4]
```diff
@@ -78,7 +78,8 @@ public class SetAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
             HelpMessage = "A list of P2S VPN client tunneling protocols")]
         [ValidateSet(
             MNM.VpnClientProtocol.SSTP,
-            MNM.VpnClientProtocol.IkeV2)]
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/UpdateAzureVirtualNetworkGatewayCommand.cs]: Change this from a ValidateSet to an ArgumentCompleter

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 5]
```diff
@@ -3,4 +3,7 @@
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Network" version="19.0.2-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/packages.config]: remove all changes to this file

**Comment by henry416** [comment]
@cormacpayne feedback addressed.
Currently uploaded nuget package does not have my change, do we have a date when SDK will be generated for June 15 release? (I checked in REST specs May 24)

**Comment by henry416** [comment]
@cormacpayne Added tests. ArgumentCompleter seems to break powershell inputs, so reverted to original.

**Comment by henry416** [comment]
@markcowl Added tests. ArgumentCompleter seems to break powershell inputs, so reverted to original.
Still need to know when June 15 SDK is generated and uploaded, otherwise can't push this.

**Comment by markcowl** [comment]
@henry416 marking as 'do not merge' until we hear from the folks managing the network sdk

**Comment by henry416** [comment]
@markcowl Can we target the June 29th release?

**Comment by cormacpayne** [comment]
@praries880 would you mind taking a look at this PR while Mark is OOF?

**Comment by cormacpayne** [comment]
@MikhailTryakhov Hey Mikhail, what's the status of this PR?

**Comment by praries880** [comment]
@cormacpayne  We are going to merge the changes into the network branch... heres the PR for that : https://github.com/Azure/azure-powershell/pull/6481.

@henry416 / @MikhailTryakhov  is it ok to close this PR?

**Comment by henry416** [comment]
@MikhailTryakhov When will we merge Network-2018-06-01 into build?

**Comment by praries880** [comment]
@MikhailTryakhov do I need to review this PR? this one was closed in favor of 
 [6481](https://github.com/Azure/azure-powershell/pull/6481)

**Comment by MikhailTryakhov** [comment]
@praries880  I'm not sure why did @henry416  closed this PR.
He owns his changes delivery so @henry416  please provide more context here

**Comment by henry416** [comment]
@MikhailTryakhov Didn't you say that by merging changes into 2018-06-01 branch it will eventually go into June 29th release? We already completed the merge for that in the linked PR https://github.com/Azure/azure-powershell/pull/6481

**Comment by MikhailTryakhov** [comment]
@henry416  I will merge all the progress from ```Network-2018-06-01``` to release tomorrow.
To have it delivered **you** must have all the code you need merged in time to the monthly branch...

**Comment by henry416** [comment]
Clarified with @MikhailTryakhov offline, this PR can remained closed. Changes were merged to Network-2018-06-01 as intended.

**Pull request closed without merging**